src/context.rs

```diff
@@ -35,20 +35,21 @@ use crate::timer;
 /// public and stable API is `ggez`'s module-level functions and
 /// types.
 pub struct Context {
-    /// Filesystem state
-    pub filesystem: Filesystem,
-    /// Graphics state
-    pub(crate) gfx_context: crate::graphics::context::GraphicsContext,
-    /// Timer state
-    pub timer_context: timer::TimeContext,
-    /// Audio context
-    pub audio_context: Box<dyn audio::AudioContext>,
-    /// Keyboard context
-    pub keyboard_context: keyboard::KeyboardContext,
-    /// Mouse context
-    pub mouse_context: mouse::MouseContext,
-    /// Gamepad context
-    pub gamepad_context: Box<dyn gamepad::GamepadContext>,
+    /// Filesystem state.
+    pub fs: Filesystem,
+    /// Graphics state.
+    pub(crate) gfx: crate::graphics::context::GraphicsContext,
+    /// Timer state.
+    pub time: timer::TimeContext,
+    /// Audio context.
+    #[cfg(feature = "audio")]
+    pub audio: audio::AudioContext,
+    /// Keyboard input context.
+    pub keyboard: input::keyboard::KeyboardContext,
+    /// Mouse input context.
+    pub mouse: input::mouse::MouseContext,
+    /// Gamepad input context.
+    pub gamepad: input::gamepad::GamepadContext,
```

Comment:
This needs the same treatment as audio, as it only compiles as long as `#[cfg(feature = "gamepad")]` is true.
```suggestion
    /// Gamepad input context.
    #[cfg(feature = "gamepad")]
    pub gamepad: input::gamepad::GamepadContext,
```